lib/scoring.rb

```diff
@@ -0,0 +1,53 @@
+class Scrabble
+  # SCORE is constant within the class-- tiles always have the same value
+  SCORE = {"a" => 1, "e" => 1, "i" => 1, "o" => 1, "u" => 1,
+           "l" => 1, "n" => 1, "r" => 1, "s" => 1, "t" => 1,
+           "d" => 2, "g" => 2, "b" => 3, "c" => 3, "m" => 3,
+           "p" => 3, "f" => 4, "h" => 4, "v" => 4, "w" => 4,
+           "y" => 4, "k" => 5, "j" => 8, "x" => 8, "q" =>10,
+           "z" => 10}
+
+  def self.score(word)
+    total_points = 0
+    # check to see that the input is actually a string (not nil or a number)
+    raise ArgumentError.new "You can only use strings" unless word.class == String
+
+    # check that the string only contains valid English letters, no numbers or symbols, controls for case
+    word_array = word.chars.map { |each| each.downcase }
+    raise ArgumentError.new "You can only use standard English letters" unless word_array.all? { |letter| SCORE[letter] }
+
+    # check that the string is between 1 and 7 letters
+    raise ArgumentError.new "You must play between 1 and 7 letters" unless word_array.length >= 1 && word_array.length <= 7
+
+    # Finally a valid word! Assigning a point value to each letter in the array
+    word_array.each do |letter|
```

Comment:
There's nothing wrong with your `each` loop here, but many methods available in Enumerable exist because of really common patterns (and are optimized to do them efficiently). The `reduce` method does exactly what you want here--compute a series of scores and then return a single value. [Here's a great blog post about using `reduce`](http://nishacodes.tumblr.com/post/80652377352/using-reduce-aka-inject-in-ruby).

With `reduce`, the entire scoring algorithm could look something like:

```ruby
return word.each_char.reduce(0) { |total, letter| total + SCORE[letter] }
```